velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp

```diff
@@ -15,178 +15,1541 @@
+      const auto valColName = fmt::format("c{}", i);
+      const std::string verifyDuckDbSql = "SELECT " + valColName +
+          " FROM tmp WHERE " + colName + " = ( SELECT " + funcName + "(" +
+          colName + ") FROM tmp) LIMIT 1";
+      SCOPED_TRACE(fmt::format(
+          "{}({}, {})\nverifyDuckDbSql: {}",
+          aggName,
+          valColName,
+          colName,
+          verifyDuckDbSql));
       auto op =
           PlanBuilder()
               .values(vectors)
               .partialAggregation(
                   {}, {fmt::format("{}({}, {})", aggName, valColName, colName)})
               .finalAggregation()
               .planNode();
+      assertQuery(op, verifyDuckDbSql);
+    }
+  }
 
-      assertQuery(
-          op,
-          "SELECT " + valColName + " FROM tmp WHERE " + colName +
-              " = ( SELECT " + funcName + "(" + colName +
-              ") FROM tmp) LIMIT 1");
+  void testGroupByAggregation(
+      const std::vector<RowVectorPtr>& vectors,
+      const std::string& aggName,
+      const std::string& colName) {
+    const std::string funcName = aggName == kMaxBy ? "max" : "min";
+    for (int volColIdx = 0; volColIdx < rowType_->size(); ++volColIdx) {
+      const int groupByColIdx = folly::Random::rand32(0, rowType_->size());
+      const std::string groupByColName = rowType_->nameOf(groupByColIdx);
+      const std::string valColName = rowType_->nameOf(volColIdx);
+      const std::string verifyDuckDbSql = "SELECT " + groupByColName + ", " +
+          aggName + "(" + valColName + ", " + colName + ") FROM tmp GROUP BY " +
+          groupByColName;
+      SCOPED_TRACE(fmt::format(
+          "{}({}, {}) GROUP BY {}\nverifyDuckDbSql: {}",
+          funcName,
+          valColName,
+          colName,
+          groupByColName,
+          verifyDuckDbSql));
+      auto op =
+          PlanBuilder()
+              .values(vectors)
+              .partialAggregation(
+                  {groupByColName},
+                  {fmt::format("{}({}, {})", aggName, valColName, colName)})
+              .finalAggregation()
+              .planNode();
+      assertQuery(op, verifyDuckDbSql);
     }
   }
 
   RowTypePtr rowType_{
-      ROW({"c0", "c1", "c2", "c3", "c4", "c5", "c6"},
+      ROW({"c0", "c1", "c2", "c3", "c4", "c5", "c6", "c7"},
           {TINYINT(),
            SMALLINT(),
            INTEGER(),
            BIGINT(),
            REAL(),
            DOUBLE(),
-           BIGINT()})};
+           BIGINT(),
+           VARCHAR()})};
 };
 
-TEST_F(MinMaxByAggregationTest, maxByPartialConst) {
-  // Const values.
-  auto vectors = {
-      makeRowVector({
-          makeConstant(5, 10),
-          makeConstant(10, 10),
-      }),
-  };
+TEST_F(MinMaxByAggregationTest, minMaxByPartialConst) {
+  for (bool isMin : {false, true}) {
+    struct {
+      const RowVectorPtr inputRowVector;
+      const std::string verifyDuckDbSql;
+      const bool expectedBuildFailure = false;
```

Comment:
Why build failure?

Reply:
It is verify that some data type we don't support it and expect an exception here when build velox operator plan.